File: projects/src/main/java/io/nirahtech/erp/projects/Team.java
package io.nirahtech.erp.projects;

import java.util.Set;

public class Team {
    private final TeamIdentifier id;
    private final String name;
    private final String description;
    private final ProjectMember manager;
    private final Set<ProjectMember> members;

    public Team(TeamIdentifier id, String name, String description, ProjectMember manager, Set<ProjectMember> members) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.manager = manager;
        this.members = members;
    }

    public TeamIdentifier getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public ProjectMember getManager() {
        return manager;
    }

    public Set<ProjectMember> getMembers() {
        return members;
    }
}

File: webapp/src/main/react/src/data/Team.ts
import ProjectMember from "./ProjectMember";
import TeamIdentifier from "./TeamIdentifier";

class Team {
  private readonly id: TeamIdentifier;
  private readonly name: string;
  private readonly description: string;
  private readonly manager: ProjectMember;
  private readonly members: Set<ProjectMember>;

  constructor(id: TeamIdentifier, name: string, description: string, manager: ProjectMember, members: Set<ProjectMember>) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.manager = manager;
    this.members = new Set<ProjectMember>(members);
  }

  getId(): TeamIdentifier {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getDescription(): string {
    return this.description;
  }

  getManager(): ProjectMember {
    return this.manager;
  }

  getMembers(): Set<ProjectMember> {
    return this.members;
  }
}

export default Team;
